lib/livebook/hubs.ex

```diff
@@ -186,18 +186,67 @@ defmodule Livebook.Hubs do
   ## Example
 
       iex> get_connected_hubs()
-      [%{pid: #PID<0.178.0>, hub: %Enterprise{}}, ...]
+      [%Enterprise{}, ...]
 
   """
-  @spec get_connected_hubs() :: connected_hubs()
+  @spec get_connected_hubs() :: list(Provider.t())
   def get_connected_hubs do
-    for hub <- get_hubs(), connected = get_connected_hub(hub), do: connected
+    for hub <- get_hubs(),
+        capability?(hub, :connect),
+        Provider.connected?(hub),
+        into: [],
+        do: hub
+  end
+
+  @doc """
+  Gets a list of connected hubs with given capabilities.
+
+  ## Example
+
+      iex> get_connected_hubs([:secrets])
+      [%Enterprise{}, ...]
+
+  """
+  @spec get_connected_hubs(Provider.capabilities()) :: list(Provider.t())
+  def get_connected_hubs(capabilities) do
+    for hub <- get_connected_hubs(),
+        capability?(hub, capabilities -- [:connect]),
+        into: [],
+        do: hub
+  end
+
+  @doc """
+  Gets a list of hub secrets.
+
+  It gets from all hubs with secret management.
+  """
+  @spec get_secrets() :: list(Secret.t())
+  def get_secrets do
+    for hub <- get_connected_hubs([:secrets]),
+        secret <- Provider.get_secrets(hub),
+        into: [],
+        do: secret
   end
 
-  defp get_connected_hub(hub) do
-    case Registry.lookup(Livebook.HubsRegistry, hub.id) do
-      [{pid, _}] -> %{pid: pid, hub: hub}
-      [] -> nil
+  @doc """
+  Creates a secret for given hub.
+  """
+  @spec create_secret(Secret.t()) :: :ok | {:error, list({String.t(), list(String.t())})}
+  def create_secret(%Secret{origin: {:hub, id}} = secret) do
+    with {:ok, hub} <- get_hub(id),
+         true <- capability?(hub, :secrets) do
+      Provider.create_secret(hub, secret)
+    else
+      :error -> {:error, %{errors: [{"hub_id", {"doest not exists", []}}]}}
+      false -> {:error, %{errors: [{"hub_id", {"is invalid", []}}]}}
     end
   end
+
+  defp capability?(hub, capabilities) when is_list(capabilities) do
+    Enum.all?(capabilities, &capability?(hub, &1))
+  end
+
+  defp capability?(hub, capability) when is_atom(capability) do
+    capability in Provider.capabilities(hub)
+  end
```

Comment:
```suggestion
  defp capability?(hub, capabilities) do
    capabilities = List.wrap(capabilities)
    capabilities -- Provider.capabilities(hub) == []
  end
```

although it would be fine to require `capabilities` to be a list in the first place :)